torch/csrc/jit/mobile/versioned_operators.cpp

```diff
@@ -0,0 +1,185 @@
+      ", supported in this runtime. Update the runtime to be compatible",
+      " to this model.");
+
+  for (const auto& info : container.version_vec) {
+    if (info.version == op_version) {
+      return info;
+    }
+  }
+
+  if (op_version == container.max_version() + 1) {
+    // Current version
+    return VersionInfo(op_version, name, c10::nullopt);
+  }
+
+  // The situation where op_version is in [min_version, max_version + 1], but
+  // not in the container, which should not happen.
+  TORCH_CHECK(
+      true,
+      "The version number of ",
+      c10::toString(name),
+      ", ",
+      op_version,
+      " is not compatible in this runtime. ");
+
+  return VersionInfo(
+      0, c10::OperatorName("aten::_convolution", ""), c10::nullopt);
+}
+
+static const std::unordered_map<std::string, VersionInfoContainer>
+    op_version_table(
+        {{"aten::_convolution",
+          VersionInfoContainer(
+              {{0, c10::OperatorName("aten::_convolution", ""), true}})}});
+} // namespace
+
+OperatorFunctor findOperatorFromName(const c10::OperatorName& opname) {
+  auto jit_op = findOperatorFor(opname);
+  OperatorFunctor fn;
+  if (jit_op) {
+    fn = [jit_op](Stack& stack) { jit_op->getOperation()(&stack); };
+  } else {
+    auto op = c10::Dispatcher::singleton().findSchema(opname);
+    if (op.has_value()) {
+      fn = [op](Stack& stack) { op->callBoxed(&stack); };
+    } else {
+      return fn;
+    }
+  }
+  return nullptr;
+}
+
+OperatorFunctor operator_resolver(
+    const c10::OperatorName& opname,
+    int64_t op_version,
+    int64_t model_version) {
+  if (model_version > 0x3LL) {
+    auto it = op_version_table.find(toString(opname));
+    if (it == op_version_table.end()) {
+      // Not in the version table, by default it fall through the original
+      // opname with no compatibility treatment
+      return findOperatorFromName(opname);
+    } else {
+      auto opinfo = getVersioninfo(it->second, opname, op_version);
+      auto fn = findOperatorFromName(opinfo.name);
+      if (opinfo.default_val) {
+        fn = [fn, opinfo](Stack& stack) {
+          stack.push_back(opinfo.default_val.value());
+          fn(stack);
+        };
+      }
+      return fn;
+    }
+  } else if (model_version == 0x3LL) {
+    auto fn = findOperatorFromName(opname);
+    if (opname == c10::OperatorName("aten::_convolution", "")) {
+      // Since byte-code versions 0x4L, convolution has an additional
+      // default-value argument (allow_tf32=True, see
+      // https://github.com/pytorch/pytorch/pull/40737). This wrapper handles
+      // backward compatibility with models of byte-code version <= 0x3L, where
+      // this bool argument does not yet exist.
+      fn = [fn](Stack& stack) {
+        stack.push_back(true);
+        fn(stack);
+      };
+    }
+    return fn;
+  }
+  return nullptr;
+}
+
+TORCH_API std::unordered_map<std::string, std::pair<int64_t, int64_t>>
+get_op_version_table() {
```

Comment:
Seems like this function could just return a pointer to the actual table?

Reply:
This function is a public API to query "the operator versioned with there [min, max] range". It's used for compatibility check. Since it does not need the details (such as default values) that the actual table contains, we provide this simpler return type.